### ISSUE TITLE
[Feature] Anima-2.9B

### ISSUE THREAD

**Issue by JohnLoveJoy** [leejet/stable-diffusion.cpp#1877]
### Feature Summary

Anima-2.9B is a fine-tune and layer-expansion of circlestone-labs/Anima. The base Anima model targets anime, illustration, and non-photorealistic art

### Detailed Description

<img width="4000" height="3000" alt="Image" src="https://github.com/user-attachments/assets/77efb914-b30e-45ea-b26a-1fe93c29153f" />

HF: https://huggingface.co/Gazingstars123/Anima-2.9B

**Comment by leejet** [comment]
It's already supported.

**Comment by JohnLoveJoy** [comment]
> It's already supported.

It doesn't seem so? SD.cpp supports standard Anima, not this expanded version.

<img width="1024" height="1024" alt="Image" src="https://github.com/user-attachments/assets/a5c296c0-51ea-4c63-9ede-5cdae6864aca" />

**Comment by stduhpf** [comment]
So it did run @JohnLoveJoy? Any errors or warnings in the logs? Because if the image looks bad it could just be either bad sampler/scheduler config or simply a poorly trained model.

**Comment by JohnLoveJoy** [comment]
> So it did run [@JohnLoveJoy](https://github.com/JohnLoveJoy)? Any errors or warnings in the logs? Because if the image looks bad it could just be either bad sampler/scheduler config or simply a poorly trained model.

I used the same sampler and scheduler I use for the standard Anima. I tried various combinations and prompts, but the result is always ruined images.

If ComfyUI requires a custom node to run the expanded version of Anima, why would SD.cpp work out of the box?

https://github.com/gazingstars123/ComfyUI-Anima-2.9B/blob/main/__init__.py

```
import comfy.model_detection

# Store the original UNet config detection function
orig_detect_unet_config = comfy.model_detection.detect_unet_config

def patched_detect_unet_config(state_dict, key_prefix, metadata=None):
    try:
        dit_config = orig_detect_unet_config(state_dict, key_prefix, metadata)
    except Exception as e:
        raise e

    try:
        if dit_config is not None and dit_config.get("image_model") == "anima":
            max_block = -1
            prefix = f"{key_prefix}blocks."

            # Scan keys to detect actual block count
            for k in state_dict.keys():
                if k.startswith(prefix):
                    parts = k[len(prefix):].split(".")
                    if parts and parts[0].isdigit():
                        max_block = max(max_block, int(parts[0]))

            if max_block != -1:
                actual_blocks = max_block + 1
                if actual_blocks != dit_config.get("num_blocks"):
                    print(f"[Anima 2.9B Patch] Dynamically patching Anima blocks from {dit_config.get('num_blocks')} to {actual_blocks}")
                    dit_config["num_blocks"] = actual_blocks
    except Exception as patch_err:
        print(f"[Anima 2.9B Patch] Warning: Failed to apply dynamic block patch: {patch_err}")

    return dit_config

# Apply patch
try:
    comfy.model_detection.detect_unet_config = patched_detect_unet_config
    print("[Anima 2.9B Patch] Successfully loaded ComfyUI model patch.")
except Exception as init_err:
    print(f"[Anima 2.9B Patch] Error: Failed to load ComfyUI model patch: {init_err}")

# ComfyUI Custom Node registrations
NODE_CLASS_MAPPINGS = {}
NODE_DISPLAY_NAME_MAPPINGS = {}
```
